File: modules/kavi-ssh/android/src/main/java/com/kavi/modules/ssh/KaviSshRecords.kt
package com.kavi.modules.ssh

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record
import expo.modules.kotlin.types.Enumerable

internal class SshEndpointRecord : Record {
  @Field
  lateinit var host: String

  @Field
  var port: Int = 0
}

internal class SshHostKeyRecord : Record {
  @Field
  lateinit var algorithm: String

  @Field
  lateinit var publicKeyBase64: String

  @Field
  lateinit var fingerprintSha256: String
}

internal enum class SshAuthenticationKind(val value: String) : Enumerable {
  PASSWORD("password"),
  PRIVATE_KEY("private-key")
}

internal class SshAuthenticationRecord : Record {
  @Field
  lateinit var kind: SshAuthenticationKind

  @Field
  var password: String? = null

  @Field
  var privateKey: String? = null

  @Field
  var passphrase: String? = null
}

internal class DiscoverHostKeyRequestRecord : Record {
  @Field
  lateinit var endpoint: SshEndpointRecord

  @Field
  var timeoutMs: Int = 0
}

internal class ConnectVerifiedRequestRecord : Record {
  @Field
  lateinit var endpoint: SshEndpointRecord

  @Field
  lateinit var username: String

  @Field
  lateinit var authentication: SshAuthenticationRecord

  @Field
  lateinit var expectedHostKey: SshHostKeyRecord

  @Field
  var timeoutMs: Int = 0
}

internal class ExecRequestRecord : Record {
  @Field
  lateinit var connectionId: String

  @Field
  lateinit var command: String

  @Field
  var timeoutMs: Int = 0

  @Field
  var outputLimitBytes: Int = 0
}

internal class DisconnectRequestRecord : Record {
  @Field
  lateinit var connectionId: String

  @Field
  var timeoutMs: Int = 0
}

File: modules/kavi-ssh/ios/KaviSshRecords.swift
import ExpoModulesCore

internal struct SshEndpointRecord: Record {
  @Field(.required) var host: String
  @Field(.required) var port: Int
}

internal struct SshHostKeyRecord: Record {
  @Field(.required) var algorithm: String
  @Field(.required) var publicKeyBase64: String
  @Field(.required) var fingerprintSha256: String
}

internal enum SshAuthenticationKind: String, Enumerable {
  case password
  case privateKey = "private-key"
}

internal struct SshAuthenticationRecord: Record {
  @Field(.required) var kind: SshAuthenticationKind = .password
  @Field var password: String?
  @Field var privateKey: String?
  @Field var passphrase: String?
}

internal struct DiscoverHostKeyRequestRecord: Record {
  @Field(.required) var endpoint: SshEndpointRecord
  @Field(.required) var timeoutMs: Int
}

internal struct ConnectVerifiedRequestRecord: Record {
  @Field(.required) var endpoint: SshEndpointRecord
  @Field(.required) var username: String
  @Field(.required) var authentication: SshAuthenticationRecord
  @Field(.required) var expectedHostKey: SshHostKeyRecord
  @Field(.required) var timeoutMs: Int
}

internal struct ExecRequestRecord: Record {
  @Field(.required) var connectionId: String
  @Field(.required) var command: String
  @Field(.required) var timeoutMs: Int
  @Field(.required) var outputLimitBytes: Int
}

internal struct DisconnectRequestRecord: Record {
  @Field(.required) var connectionId: String
  @Field(.required) var timeoutMs: Int
}
